tensorflow/examples/tutorials/word2vec/word2vec_basic.py

```diff
@@ -102,6 +110,9 @@ def generate_batch(batch_size, num_skips, skip_window):
   span = 2 * skip_window + 1  # [ skip_window target skip_window ]
   buffer = collections.deque(maxlen=span)
   for _ in range(span):
+    if len(buffer)!=0 and data_index == 0:
+      skip_to_start(buffer, skip_window)
+      break
     buffer.append(data[data_index])
     data_index = (data_index + 1) % len(data)
```

Comment:
Please use the shorter and clearer version.  Note that you don't need to clear the buffer for the first use, since it was created immediately before.  However, note that my shorter clearer version is slightly wrong.  It should be

    if data_index + span > len(data):
      data_index = 0
    buffer.extend(data[data_index:data_index + span])
    data_index += span